### DISCUSSION TITLE
Sequence of the Amino Acid descriptors in p2rank

### DISCUSSION THREAD

**Discussion by Sghl1** [rdk/p2rank#64] [Q&A] [answered]
Dear Community, 

I am actually working on research regarding the performance of using p2rank to identify protein binding site prediction, 

keep on my eye that the results obtained for the pockets sequence in the amino acid descriptors position calculated are not identically  similar in position as in RCSB Protein Data Bank, it shows me that the sequence has been cut from the beginning, 

There is any explanation for the reason why this is happening? 

it is because of the algorithm? 

Thank you 

Regards 

Sghl

**Comment by rdk** [comment]
Hi @Sghl1, thank you for your question. 

Could you provide one example where that happens? Could you also elaborate on what do you exactly mean by "results obtained for the pockets sequence in the amino acid descriptors position calculated"? Which output file are you looking at?

**Comment by Sghl1** [comment]
Dear Rdk,

Please see the screenshots attached, showing the sequence of the PDB code
interested in P2RANK and in https://www.rcsb.org/sequence/1O86 sequence of
the same protein.

The amino acid descriptor in Pocket 1 shows in the sequence that is ( E )
and the position is (123) while in Rcsb the sequence of the same protein
shows that is in the 126 position

Why has the position of the sequence been cut 3 places from the beginning?

it is relative because of the algorithm?

Thank you

Kind regards

Sghl




On Thu, 20 Jul 2023 at 08:55, rdk ***@***.***> wrote:

> Hi @Sghl1 <https://github.com/Sghl1>, thank you for your question.
>
> Could you provide one example where that happens? Could you aleo elaborate
> on what do you exactly mean by "results obtained for the pockets sequence
> in the amino acid descriptors position calculated"? Which output file are
> you looking at?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdk/p2rank/discussions/64#discussioncomment-6495841>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBMGIPT36FCPDZ4URKX4QD3XRDI4HANCNFSM6AAAAAA2QLFPLY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 


This
message and its attachments are private and confidential. If you 
have received
this message in error, please notify the sender and remove it 
and its
attachments from your system.

The University of Westminster is a 
charity and a company limited by guarantee. Registration number: 977818 
England. Registered Office: 309 Regent Street, London W1B 2UW.

**Comment by rdk** [comment]
@Sghl1 I don't see any screenshots. 

> The amino acid descriptor in Pocket 1 shows in the sequence that is ( E )
> and the position is (123)

Where do you see that?

**Reply by Sghl1** [reply]
<img width="1227" alt="Screenshot 2023-07-20 at 09 19 13" src="https://github.com/rdk/p2rank/assets/140010558/18175385-b0e6-4d7b-84b4-091df27d8576">

**Comment by Sghl1** [comment]
[
<img width="1401" alt="Screenshot 2023-07-20 at 09 09 57" src="https://github.com/rdk/p2rank/assets/140010558/d96d865e-f236-477a-aa85-e1c5dd3508f2">
[](url)]([url](<img width="1227" alt="Screenshot 2023-07-20 at 09 19 13" src="https://github.com/rdk/p2rank/assets/140010558/85a77d1a-7ee8-426a-944a-dd5256e098b9">))

**Comment by rdk** [comment, marked as answer]
@Sghl1 in this case the root of a problem seems to be in the fact that not all residues of the uniprot sequence are modelled (i.e. present in the PDB structure), which is a common occurence in the PDB. Another fact is that the sequence visualizers in both https://www.rcsb.org/sequence/1O86 and https://prankweb.cz/ show on the top a line with numbers that represent a simple position in the sequence from 1 to n.

The sequence visualization in PDB https://www.rcsb.org/sequence/1O86 displays the Uniprot sequence overlaid with PDB sequence and first 3 residues are not present in the PDB sequence (see UNMODELED row).

I'd recommend to disregard whatever sequence numbers are displayed in the sequence visualisers and instead look at so called author sequence numbers of individual residues (i.e. numbers that are present in pdb/cif files and can serve as unique identifiers of a given residue (together with a chain name and insertion code)). You can download results from prankweb.cz using "Download data" button and look into the file `structure.pdb_residues.csv` and compare the residue numbering there with the one in the PDB or cif file.

**Reply by Sghl1** [reply]
Dear Idk, 

Thank you for your replay, 
Was very helpful, 

it is possible for me to add your recommendation given me by GitHub into my research to disregard the sequence numbers displayed in the sequence and only look at the download data (structure.pdb_residues.csv) and reference you? 

Thanks 

Sghl
